### PR TITLE
SDKTECHNO-182 - Adding Spark 3.1 with AWS connectors

### PR DESCRIPTION
**Pull request by tquiviger**
Adding Spark 3.1 with AWS connectors (S3)

### CODE REVIEW

File: technologies/job/spark/spark-aws-3.1/innerContexts/jre/spark-aws-3.1-jre-11/build.me [hunk 1]
```diff
@@ -0,0 +1 @@
+08/07/2022
```
Comment (guillaumeprevost76) [inline technologies/job/spark/spark-aws-3.1/innerContexts/jre/spark-aws-3.1-jre-11/build.me:1]: What's the meaning of this date ? 
if it's the creation of the image by us, why is it not the current date (20/09/2022)?

File: technologies/job/spark/spark-aws-3.1/innerContexts/python/spark-aws-3.1-python-3.7/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import com.saagie.technologies.getVersionForDocker
+
+
+apply<DockerRemoteApiPlugin>()
+apply<SaagieTechnologiesGradlePlugin>()
+
+val dockerInfo = readDockerInfo(projectDir)
+
+tasks.withType(com.bmuschko.gradle.docker.tasks.image.DockerBuildImage::class) {
+    dependsOn(":spark-3.1:testImage")
```
Comment (guillaumeprevost76) [inline technologies/job/spark/spark-aws-3.1/innerContexts/python/spark-aws-3.1-python-3.7/build.gradle.kts:30]: Should it be "spark-aws-3.1:testImage" ?

File: technologies/job/spark/spark-aws-3.1/innerContexts/python/spark-aws-3.1-python-3.7/build.me [hunk 3]
```diff
@@ -0,0 +1 @@
+08/07/2022
```
Comment (guillaumeprevost76) [inline technologies/job/spark/spark-aws-3.1/innerContexts/python/spark-aws-3.1-python-3.7/build.me:1]: What's the meaning of this date ?
if it's the creation of the image by us, why is it not the current date (20/09/2022)?

File: technologies/job/spark/spark-aws-3.1/innerContexts/python/spark-aws-3.1-python-3.8/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import com.saagie.technologies.getVersionForDocker
+
+
+apply<DockerRemoteApiPlugin>()
+apply<SaagieTechnologiesGradlePlugin>()
+
+val dockerInfo = readDockerInfo(projectDir)
+
+tasks.withType(com.bmuschko.gradle.docker.tasks.image.DockerBuildImage::class) {
+    dependsOn(":spark-3.1:testImage")
```
Comment (guillaumeprevost76) [inline technologies/job/spark/spark-aws-3.1/innerContexts/python/spark-aws-3.1-python-3.8/build.gradle.kts:30]: Should it be "spark-aws-3.1:testImage" ?

File: technologies/job/spark/spark-aws-3.1/innerContexts/python/spark-aws-3.1-python-3.8/build.me [hunk 5]
```diff
@@ -0,0 +1 @@
+04/11/2021
```
Comment (guillaumeprevost76) [inline technologies/job/spark/spark-aws-3.1/innerContexts/python/spark-aws-3.1-python-3.8/build.me:1]: What's the meaning of this date ?
if it's the creation of the image by us, why is it not the current date (20/09/2022)?

File: technologies/job/spark/spark-aws-3.1/innerContexts/python/spark-aws-3.1-python-3.9/build.gradle.kts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+import com.saagie.technologies.getVersionForDocker
+
+
+apply<DockerRemoteApiPlugin>()
+apply<SaagieTechnologiesGradlePlugin>()
+
+val dockerInfo = readDockerInfo(projectDir)
+
+tasks.withType(com.bmuschko.gradle.docker.tasks.image.DockerBuildImage::class) {
+    dependsOn(":spark-3.1:testImage")
```
Comment (guillaumeprevost76) [inline technologies/job/spark/spark-aws-3.1/innerContexts/python/spark-aws-3.1-python-3.9/build.gradle.kts:30]: Should it be "spark-aws-3.1:testImage" ?

File: technologies/job/spark/spark-aws-3.1/innerContexts/python/spark-aws-3.1-python-3.9/build.me [hunk 7]
```diff
@@ -0,0 +1 @@
+04/11/2021
```
Comment (guillaumeprevost76) [inline technologies/job/spark/spark-aws-3.1/innerContexts/python/spark-aws-3.1-python-3.9/build.me:1]: What's the meaning of this date ?
if it's the creation of the image by us, why is it not the current date (20/09/2022)?

**Review by Qiwei08 — APPROVED** [review]

File: technologies/job/spark/spark-aws-3.1/innerContexts/jre/spark-aws-3.1-jre-11/build.me (hunk 1 above)
Comment (tquiviger) [reply technologies/job/spark/spark-aws-3.1/innerContexts/jre/spark-aws-3.1-jre-11/build.me:1]: It's a hack when you want to force the build, not necessary on this PR I can delete the file

File: technologies/job/spark/spark-aws-3.1/innerContexts/python/spark-aws-3.1-python-3.9/build.gradle.kts (hunk 6 above)
Comment (tquiviger) [reply technologies/job/spark/spark-aws-3.1/innerContexts/python/spark-aws-3.1-python-3.9/build.gradle.kts]: indeed

File: technologies/job/spark/spark-aws-3.1/innerContexts/python/spark-aws-3.1-python-3.7/build.gradle.kts (hunk 2 above)
Comment (tquiviger) [reply technologies/job/spark/spark-aws-3.1/innerContexts/python/spark-aws-3.1-python-3.7/build.gradle.kts]: indeed

File: technologies/job/spark/spark-aws-3.1/innerContexts/python/spark-aws-3.1-python-3.7/build.me (hunk 3 above)
Comment (tquiviger) [reply technologies/job/spark/spark-aws-3.1/innerContexts/python/spark-aws-3.1-python-3.7/build.me]: It's a hack when you want to force the build, not necessary on this PR I can delete the file

File: technologies/job/spark/spark-aws-3.1/innerContexts/python/spark-aws-3.1-python-3.8/build.gradle.kts (hunk 4 above)
Comment (tquiviger) [reply technologies/job/spark/spark-aws-3.1/innerContexts/python/spark-aws-3.1-python-3.8/build.gradle.kts]: indeed

File: technologies/job/spark/spark-aws-3.1/innerContexts/python/spark-aws-3.1-python-3.8/build.me (hunk 5 above)
Comment (tquiviger) [reply technologies/job/spark/spark-aws-3.1/innerContexts/python/spark-aws-3.1-python-3.8/build.me]: It's a hack when you want to force the build, not necessary on this PR I can delete the file

File: technologies/job/spark/spark-aws-3.1/innerContexts/python/spark-aws-3.1-python-3.9/build.me (hunk 7 above)
Comment (tquiviger) [reply technologies/job/spark/spark-aws-3.1/innerContexts/python/spark-aws-3.1-python-3.9/build.me]: It's a hack when you want to force the build, not necessary on this PR I can delete the file

**Review by guillaumeprevost76 — APPROVED** [review]

**Pull request merged**